lib/rubocop/cop/layout/empty_lines_around_access_modifier.rb

```diff
@@ -89,8 +90,10 @@ def block_start?(line)
           line.match(/ (do|{)( \|.*?\|)?\s?$/)
         end
 
-        def body_end?(line)
-          line =~ /^\s*end/
+        def last_children?(node)
```

Comment:
I guess you meant to write `last_child?` here. While I don't like regexp matches a lot, I don't think your approach is much better. I guess you could have just fixed the regexp.

Reply:
@bbatsov we have small discuss with @pocke [here](https://github.com/bbatsov/rubocop/pull/5522#discussion_r165286303)

While regexp might work, this approach with `last_child` seems to be right and always valid.

However, you are the boss :) If you think that regex is better - I can use it, but I don't get your point right now - what is wrong with `last_child`?

Reply:
Mostly the name of the method. `children` is plural. And a few methods look a bit odd with the extra param - e.g. `next_line_empty?(node, line)`. It's not immediately clear which `node` is that. I'm fine with the approach in general, I just want to code to be made a bit clearer.